scripts/safe_template_linter.py

```diff
@@ -1955,6 +1950,291 @@ def _find_mako_expressions(self, mako_template):
+    """
+
+    def __init__(self):
+        """
+        Init method.
+        """
+        super(PythonLinter, self).__init__()
+        self._skip_python_dirs = self._skip_dirs + ('tests',)
+
+    def process_file(self, directory, file_name):
+        """
+        Process file to determine if it is a Python file and
+        if it is safe.
+
+        Arguments:
+            directory (string): The directory of the file to be checked
+            file_name (string): A filename for a potential JavaScript file
+
+        Returns:
+            The file results containing any violations.
+
+        """
+        file_full_path = os.path.normpath(directory + '/' + file_name)
+        results = FileResults(file_full_path)
+
+        if not results.is_file:
+            return results
+
+        if file_name.lower().endswith('.py') is False:
+            return results
+
+        # don't lint this file
+        if file_name == os.path.basename(__file__):
+            return results
+
+        if not self._is_valid_directory(self._skip_python_dirs, directory):
+            return results
+
+        return self._load_and_check_file_is_safe(file_full_path, self.check_python_file_is_safe, results)
+
+    def check_python_file_is_safe(self, file_contents, results):
+        """
+        Checks for violations in a Python file.
+
+        Arguments:
+            file_contents: The contents of the JavaScript file.
+            results: A file results objects to which violations will be added.
+
+        """
+        self._check_concat_with_html(file_contents, Rules.python_concat_html, results)
+        self._check_deprecated_display_name(file_contents, results)
+        self._check_custom_escape(file_contents, results)
+        self._check_html(file_contents, results)
+        results.prepare_results(file_contents, line_comment_delim='#')
+
+    def _check_deprecated_display_name(self, file_contents, results):
+        """
+        Checks that the deprecated display_name_with_default_escaped is not
+        used. Adds violation to results if there is a problem.
+
+        Arguments:
+            file_contents: The contents of the Python file
+            results: A list of results into which violations will be added.
+
+        """
+        for match in re.finditer(r'\.display_name_with_default_escaped', file_contents):
+            expression = Expression(match.start(), match.end())
+            results.violations.append(ExpressionRuleViolation(
+                Rules.python_deprecated_display_name, expression
+            ))
+
+    def _check_custom_escape(self, file_contents, results):
+        """
+        Checks for custom escaping calls, rather than using a standard escaping
+        method.
+
+        Arguments:
+            file_contents: The contents of the Python file
+            results: A list of results into which violations will be added.
+
+        """
+        for match in re.finditer(r"""replace\(['"]<['"],""", file_contents):
```

Comment:
If this method is to be comprehensive, it'll need to check for more than strictly `replace('<',`. See https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/util/misc.py for 2 such examples:
- `replace(var`, where `var` can contain `<`
- `re.sub(...)`